internal/controller/dnsrecord_controller.go

```diff
@@ -106,40 +107,48 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
-	if err != nil {
-		reason := "ManagedZoneError"
-		message := fmt.Sprintf("The managedZone could not be loaded: %v", err)
-		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionFalse, reason, message)
-		return r.updateStatus(ctx, previous, dnsRecord, false, err)
-	}
-
+	// Update the logger with appropriate record/zone metadata
 	logger = log.FromContext(ctx).
+		WithValues("rootHost", dnsRecord.Spec.RootHost).
 		WithValues("ownerID", dnsRecord.Status.OwnerID).
-		WithValues("zoneID", managedZone.Status.ID)
+		WithValues("zoneID", dnsRecord.Status.ZoneID).
+		WithValues("zoneDomainName", dnsRecord.Status.ZoneDomainName)
 	ctx = log.IntoContext(ctx, logger)
 
 	if dnsRecord.DeletionTimestamp != nil && !dnsRecord.DeletionTimestamp.IsZero() {
 		logger.Info("Deleting DNSRecord")
-		if err = r.ReconcileHealthChecks(ctx, dnsRecord, managedZone); client.IgnoreNotFound(err) != nil {
-			return ctrl.Result{}, err
-		}
-		hadChanges, err := r.deleteRecord(ctx, dnsRecord, managedZone)
-		if err != nil {
-			logger.Error(err, "Failed to delete DNSRecord")
-			return ctrl.Result{}, err
-		}
-		// if hadChanges - the deleteRecord has successfully applied changes
-		// in this case we need to queue for validation to ensure DNS Provider retained changes
-		// before removing finalizer and deleting the DNS Record CR
-		if hadChanges {
-			return ctrl.Result{RequeueAfter: randomizedValidationRequeue}, nil
+		if dnsRecord.Status.ZoneID != "" && dnsRecord.Status.ZoneDomainName != "" {
+			// Create a dns provider with config calculated for the current dns record status (Last successful)
+			dnsProvider, err := r.ProviderFactory.ProviderFor(ctx, dnsRecord, provider.Config{
+				ZoneIDFilter:   externaldnsprovider.NewZoneIDFilter([]string{dnsRecord.Status.ZoneID}),
+				DomainFilter:   externaldnsendpoint.NewDomainFilter([]string{dnsRecord.Status.ZoneDomainName}),
+				ZoneTypeFilter: externaldnsprovider.NewZoneTypeFilter(""),
+			})
+			if err != nil {
+				logger.Error(err, "Failed to load DNS Provider")
+				reason := "DNSProviderError"
+				message := fmt.Sprintf("The dns provider could not be loaded: %v", err)
+				setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionFalse, reason, message)
+				return r.updateStatus(ctx, previous, dnsRecord, false, err)
+			}
+
+			if err = r.ReconcileHealthChecks(ctx, dnsRecord); client.IgnoreNotFound(err) != nil {
+				return ctrl.Result{}, err
+			}
+			hadChanges, err := r.deleteRecord(ctx, dnsRecord, dnsProvider)
+			if err != nil {
+				logger.Error(err, "Failed to delete DNSRecord")
+				return ctrl.Result{}, err
+			}
+			// if hadChanges - the deleteRecord has successfully applied changes
+			// in this case we need to queue for validation to ensure DNS Provider retained changes
+			// before removing finalizer and deleting the DNS Record CR
+			if hadChanges {
```

Comment:
this flow and letting it fall through to https://github.com/Kuadrant/dns-operator/pull/203/files#diff-adacb6a4589c65c2d853c0f59ca02d6b8fb50bb388b41de1d13e99b0d128cfe1R173 seem the same? Wondering if it could be removed as it will aways result in same result?

Reply:
Oh actually never mind